bika/lims/tests/doctests/ARAnalysesField.rst

```diff
@@ -0,0 +1,603 @@
+    '30.00'
+
+Created Analyses inherit that price:
+
+    >>> new_analyses = field.set(ar, all_services)
+
+    >>> analysis1 = ar[analysisservice1.getKeyword()]
+    >>> analysis2 = ar[analysisservice2.getKeyword()]
+    >>> analysis3 = ar[analysisservice3.getKeyword()]
+
+    >>> analysis1.getPrice()
+    '10.00'
+
+    >>> analysis2.getPrice()
+    '20.00'
+
+    >>> analysis3.getPrice()
+    '30.00'
+
+The `setter` also allows to set custom prices for the Analyses:
+
+    >>> prices = {
+    ...     analysisservice1.UID(): "100",
+    ...     analysisservice2.UID(): "200",
+    ...     analysisservice3.UID(): "300",
+    ... }
+
+Now we set the field with all analyses services and new prices:
+
+    >>> new_analyses = field.set(ar, all_services, prices=prices)
+
+The Analyses have now the new prices:
+
+    >>> analysis1.getPrice()
+    '100.00'
+
+    >>> analysis2.getPrice()
+    '200.00'
+
+    >>> analysis3.getPrice()
+    '300.00'
+
+The Services should retain the old prices:
+
+    >>> analysisservice1.getPrice()
+    '10.00'
+
+    >>> analysisservice2.getPrice()
+    '20.00'
+
+    >>> analysisservice3.getPrice()
+    '30.00'
+
+
+Calculations and Interim Fields
+...............................
+
+When an Analysis is assigned to an AR, it inherits its Calculation and Interim Fields.
+
+Create some interim fields:
+
+    >>> interim1 = {"keyword": "A", "title": "Interim A", "value": 1, "hidden": False, "type": "int", "unit": "x"}
+    >>> interim2 = {"keyword": "B", "title": "Interim B", "value": 2, "hidden": False, "type": "int", "unit": "x"}
+    >>> interim3 = {"keyword": "C", "title": "Interim C", "value": 3, "hidden": False, "type": "int", "unit": "x"}
+    >>> interim4 = {"keyword": "D", "title": "Interim D", "value": 4, "hidden": False, "type": "int", "unit": "x"}
+
+Append interim field `A` to the `Total Hardness` Calculation:
+
+    >>> calc4.setInterimFields([interim1])
+    >>> map(lambda x: x["keyword"], calc4.getInterimFields())
+    ['A']
+
+Append interim field `B` to the `Total Hardness` Analysis Service:
+
+    >>> analysisservice4.setInterimFields([interim2])
+    >>> map(lambda x: x["keyword"], analysisservice4.getInterimFields())
+    ['B']
+
+Now we assign the `Total Hardness` Analysis Service:
+
+    >>> new_analyses = field.set(ar, [analysisservice4])
+    >>> analysis = new_analyses[0]
+    >>> analysis
+    <Analysis at /plone/clients/client-1/water-0001-R01/THCaCO3>
+
+The created Analysis has the same Calculation attached, as the Analysis Service:
+
+    >>> analysis_calc = analysis.getCalculation()
+    >>> analysis_calc
+    <Calculation at /plone/bika_setup/bika_calculations/calculation-4>
+
+And therofore, also the same Interim Fields as the Calculation:
+
+    >>> map(lambda x: x["keyword"], analysis_calc.getInterimFields())
+    ['A']
+
+The Analysis also inherits the Interim Fields of the Analysis Service:
+
+    >>> map(lambda x: x["keyword"], analysis.getInterimFields())
+    ['B']
```

Comment:
But I don't want to rely on `Calculation` to get the interim fields from the analysis. So, I'd expect the following:
```python
    >>> map(lambda x: x["keyword"], analysis.getInterimFields())
    ['A', 'B']
```
I always assume that `analysis.getInterimFields()`always return the interim fields that must be used. I don't care about what Calculation says on this regard, because interim fields from analysis always have priority over the ones that are defined in Calculation.